File: Editorial/86_Partition_List/JavaScript/1_Two_Pointer_Approach.js
var partition = function (head, x) {
    var before_head = new ListNode(0);
    var before = before_head;
    var after_head = new ListNode(0);
    var after = after_head;
    while (head != null) {
        if (head.val < x) {
            before.next = head;
            before = before.next;
        } else {
            after.next = head;
            after = after.next;
        }
        head = head.next;
    }
    after.next = null;
    before.next = after_head.next;
    return before_head.next;
};

File: Editorial/0000-0100/0086_Partition_List/Go/1_Two_Pointer_Approach.go
func partition(head *ListNode, x int) *ListNode {
    before_head := &ListNode{0, nil}
    before := before_head
    after_head := &ListNode{0, nil}
    after := after_head
    for head != nil {
        if head.Val < x {
            before.Next = head
            before = before.Next
        } else {
            after.Next = head
            after = after.Next
        }
        head = head.Next
    }
    after.Next = nil
    before.Next = after_head.Next
    return before_head.Next
}
